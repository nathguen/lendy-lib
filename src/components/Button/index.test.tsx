import { fireEvent, render, screen } from '@testing-library/react-native';
import React, { useState } from 'react';

import { Button, Wrapper, Text, ButtonType, ButtonTypes } from '../';

const buttonText = 'Click me';
const renderedText = 'Button pressed';

const TestButton = ({ type }: { type: ButtonType }) => {
  const [buttonPressed, setButtonPressed] = useState(false);

  return (
    <Wrapper>
      <Button
        type={type}
        title={buttonText}
        onPress={() => setButtonPressed(true)}
      />

      {!!buttonPressed && <Text>{renderedText}</Text>}
    </Wrapper>
  );
};

describe('Button', () => {
  Object.values(ButtonTypes).forEach((type) => {
    it(`renders and presses ${type} Button`, () => {
      const baseContainer = render(<TestButton type={type} />);

      expect(baseContainer).toMatchSnapshot();
      expect(screen.getByText(buttonText)).toBeTruthy();

      fireEvent.press(screen.getByText(buttonText));
      expect(screen.getByText(renderedText, { exact: true })).toBeTruthy();
    });
  });

  describe('disabled state', () => {
    Object.values(ButtonTypes).forEach((type) => {
      it(`renders ${type} Button and prevents presses`, () => {
        const baseContainer = render(<TestButton type={type} />);

        expect(baseContainer).toMatchSnapshot();
        expect(screen.getByText(buttonText)).toBeTruthy();

        fireEvent.press(screen.getByText(buttonText));
      });
    });
  });
});
