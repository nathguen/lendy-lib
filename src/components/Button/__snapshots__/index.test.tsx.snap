// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled state renders primary Button and prevents presses 1`] = `
<View
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#F3ECD1",
      "borderColor": "#F3ECD1",
      "borderRadius": 30,
      "borderWidth": 3,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button disabled state renders secondary Button and prevents presses 1`] = `
<View
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderColor": "#BCB7A3",
      "borderRadius": 30,
      "borderWidth": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#BCB7A3",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button disabled state renders tertiary Button and prevents presses 1`] = `
<View
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "borderRadius": 30,
      "borderWidth": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#BCB7A3",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button renders and presses primary Button 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#191919",
      "borderColor": "#191919",
      "borderRadius": 30,
      "borderWidth": 3,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button renders and presses primary button 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "#191919",
      "borderColor": "#191919",
      "borderRadius": 30,
      "borderWidth": 3,
      "elevation": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFFFFF",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button renders and presses secondary Button 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderColor": "#191919",
      "borderRadius": 30,
      "borderWidth": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#191919",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;

exports[`Button renders and presses tertiary Button 1`] = `
<View
  accessibilityState={
    Object {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    Object {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "flex-start",
      "backgroundColor": "transparent",
      "borderColor": "transparent",
      "borderRadius": 30,
      "borderWidth": 3,
      "flexDirection": "row",
      "justifyContent": "center",
      "paddingHorizontal": 44,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#191919",
        "fontSize": 17,
        "fontWeight": "400",
        "lineHeight": 22,
        "textAlign": "center",
      }
    }
  >
    Click me
  </Text>
</View>
`;
